packages/solana-wallet-adapter/src/SolanaWalletAdapterProvider.tsx

```diff
@@ -53,24 +57,18 @@ export default function SolanaWalletAdapterProvider({
   );
 
   useEffect(() => {
-    if (wallet?.adapter.publicKey) {
-      const chain = chains
-        .filter((c) => c.type === SupportedChains.Solana)
-        .filter((c) => c.rpcEndpoint === connection?.rpcEndpoint);
+    const storedChain = get(LOCAL_STORAGE_KEY);
+    if (connected && storedChain) {
+      setSelectedChain(storedChain);
+    }
+  }, [get, connected]);
 
-      if (selectedChain?.name !== chain[0]?.name) {
-        setSelectedChain(chain[0]);
-        return;
-      }
+  useEffect(() => {
```

Comment:
do we not also need a useEffect to clear the stored chain if the user disconnects?